gap/examples.gi

```diff
@@ -370,3 +370,46 @@ GeneralisedPetersenGraphCons);
 
 InstallMethod(GeneralisedPetersenGraph, "for integer, integer", [IsInt, IsInt],
 {n, k} -> GeneralisedPetersenGraphCons(IsImmutableDigraph, n, k));
+
+InstallGlobalFunction(DIGRAPHS_HaarGraph,
+function(n)
+  local m, binaryList, D, i, j;
+  if n < 1 and IsInt(n) then
```

Comment:
This will attempt to evaluate n < 1 before IsInt(n) which can still return an error if n is incomparable with 1, if the order is changed to check IsInt(n) first this would stop any such error.  I believe you would also need to negate the check for IsInt and likely change the and to an or as you with to return the error for when n is incomparable with 1 or when n < 1.

Reply:
You may want to use `IsPosInt` here - see https://www.gap-system.org/Manuals/doc/ref/chap14.html#X82A854757DFA9C76

Reply:
Right, that makes sense. IsPosInt() seems like a good way to go.